homeassistant/components/camera/__init__.py

```diff
@@ -543,7 +543,10 @@ async def handle(self, request: web.Request, camera: Camera) -> web.StreamRespon
         """Serve camera stream, possibly with interval."""
         interval_str = request.query.get("interval")
         if interval_str is None:
-            return await camera.handle_async_mjpeg_stream(request)
+            stream = await camera.handle_async_mjpeg_stream(request)
+            if stream is None:
+                raise web.HTTPBadGateway()
```

Comment:
This should be a 503?

Reply:
I assume the majority usecase is the camera entity proxying a video feed from elsewhere, and so the error condition was more likely to be related to that upstream (this is the case  in my [example usecase](https://github.com/home-assistant/core/pull/46516/files#diff-c656a9d2f40f5a2536a3f894fd4acaa9ba6d7ef2a62d6e59a9b19845afdb7882R216)):

```
10.5.3 502 Bad Gateway
The server, while acting as a gateway or proxy, received an invalid response from the upstream server it accessed in attempting to fulfill the request.

10.5.4 503 Service Unavailable
The server is currently unable to handle the request due to a temporary overloading or maintenance of the server. The implication is that this is a temporary condition which will be alleviated after some delay.
```

But either works. You still prefer 503?

Reply:
502 is better 👍